index.js

```diff
@@ -188,6 +192,21 @@ const authenticateWithIamRole = co.wrap(function * (context, iamPrincipalArn, re
   return token
 })
 
+const getEcsMetadata = co.wrap(function * (context) {
+  context.log('getting ecs metadata')
+  let metadataResponse = yield request({ url: ecsMetadataUrl, json: true })
+  context.log(metadataResponse)
+  let data = metadataResponse.data
+  if (!data || data.Code !== 'Success') throw new Error(data)
+  context.log('ecs metadata', data)
+  let arn = data.InstanceProfileArn.split(':')
+  return {
+    accountId: arn[4],
+    roleName: context.ecsTaskRoleName,
```

Comment:
Does the role name need to be passed in by the user? Can it be derived from metadata as well?

Reply:
@tyrsius @sdford @fieldju - good question and actually why I marked this as 'WIP'. So neither the metadata container file nor the metadata task endpoint for ECS provides the IAM role afaik. Seems simplest to just have a consuming app provide this via the Cerberus config. Thoughts?

Reply:
The metadata task endpoint provides a "RoleArn" if I remember correctly which is the task execution role. I did something similar for Java and the code is [here](https://github.com/Nike-Inc/cerberus-java-client/commit/e1bea7513ded6a11a3e4097abfd5864d50fb89a7).

Reply:
@mayitbeegh That looks like you are parsing the ecs role out of the metadata url, am I reading that wrong?

Reply:
Correct. For some reason AWS has decided not to [document it.](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint.html#task-metadata-response)

Reply:
@mayitbeegh aren't we looking for the IAM role name rather than ARN though? Are you doing a lookup of the name via the cli/sdk after or am I misunderstanding your comment?

Reply:
Oh, okay that makes things super easy then. I'll update shortly

Reply:
@sdford @mayitbeegh circling back.. the IAM role name is not in the ECS task metadata endpoint (http://169.254.170.2/v2/metadata) for me, only a TaskArn